Create eLTERMetadataSchema.json

maab opened the pull request:
Introduction of a single file JSON Schema representation of the eLTER Metadata Schema

maab committed:
Create eLTERMetadataSchema.json

Introduction of a single file JSON Schema representation of the eLTER Metadata Schema

maab committed:
Move eLTERMetadataSchema.json to schemas subdirectory

maab committed:
Adding authors section to eLTERMetadataSchema.json

oggioniale reviewed (commented):
{"givenName":  "Hanna", "familyName":  "Koivula", "affiliation": "CSC - IT"},
{"givenName":  "Christoph", "familyName":  "Wohner,", "affiliation": "EEA"},
{"givenName":  "Holger", "familyName":  "Villwock", "affiliation": "SLU"},

Commit:
New authors and preparation to have multiple schema files

Commit:
Split schema into subschemas

maab committed:
Create eLTERMetadataSchemaExternalSources.json

maab committed:
Rename eLTERMetadataSchemaDarIngest.json to eLTERMetadataSchemaDatasets.json

maab committed:
Rename eLTERMetadataSchemaExternalSources.json to eLTERMetadataSchemaExternalDatasets.json

1O committed:
append blank line to each .json

to silence "incomplete last line" warning/error from AJV jsonschema validator

1O committed:
correct keywords

unknown keyword "$0id-comment" replaced with "$comment"
unknown keyword "authors" replaced with "x-authors" (`x-` indicating a custom annotation)

1O committed:
correct keywords

replace unknown keyword "_comment_on_$id" with "$comment"
replace unknown keyword "authors" with "x-authors", `x-` signalizing a custom annotation

1O committed:
add datasets schema as monolith

include all local subschemas into main schema

1O committed:
fix properties required from habitatReferences

replace `"required": true` with `"required": ["soHabitatCode", "soHabitatURI"]`

1O committed:
remove eLTERMetadataSchemaExternalDatasets.json from wrong directory

1O committed:
correct keywords and requirements notation

replace `_comment_on_$id` with `$comment`
replace `"required": true` with `"required": ["soHabitatCode", "soHabitatURI"]`

1O committed:
add monolithic schema for external datasets

pull in all referenced subschemas to obtain a monolithic schema without references to subschemas

1O committed:
add utilities for JSONschema construction

1O committed:
prefix custom keys with "x-"

1O committed:
unify $ids

following the pattern   "$id": "elter://metadata.schemas/subschemas/<filename>"
note: the id does not have to resolve to an URL on the internet; I chose this to prevent the validator from downloading stale versions hosted on GitHub, though.

1O committed:
Delete docs/new_docs/validate.R

didn't belong here

1O committed:
Create README.md

1O committed:
Update README.md
